lib/taste_tester/client.rb

```diff
@@ -136,53 +136,48 @@ def populate(stream, writer, path, destination)
       # everything is relative to the repo dir. chdir makes handling all the
       # paths within this simpler
       Dir.chdir(full_path) do
-        Find.find('.') do |p|
-          # ignore current directory. The File.directory? would also skip it,
-          # but we need to do it early because the string is too short for the
-          # next statement.
-          next if p == '.'
-          # paths are enumerated as relative to the input path '.', so we get
-          # './dir/file'. Stripping off the first two characters gives us a
-          # a cleaner 'dir/file' path.
-          relative_name = p[2..-1]
-          # some exclusions are rooted relative to cookbooks which are one
-          # level below the directory included here. Strip off the first part
-          # of the path. e.g. `fb_cookbook/spec/foo.rb` would be `spec/foo.rb`
-          # which matches `spec/*`.
-          relative_name_minus_first = relative_name.split(
-            File::SEPARATOR,
-          )[1..-1].join(File::SEPARATOR)
-          next if chefignores.ignored?(relative_name) ||
-            chefignores.ignored?(relative_name_minus_first)
-          name = File.join(destination, relative_name)
-          if File.directory?(p)
-            # skip it. This also handles symlinks to directories which aren't
-            # useful either.
-          elsif File.symlink?(p)
-            # tar handling of filenames > 100 characters gets complex. We'd use
-            # split_name from Minitar, but it's a private method. It's
-            # reasonable to assume that all symlink names in the bundle are
-            # less than 100 characters long. Long term, the version of minitar
-            # in chefdk should be upgraded.
-            fail 'Add support for long symlink paths' if name.size > 100
-            # The version of Minitar included in chefdk does not support
-            # symlinks directly. Therefore we use direct writes to the
-            # underlying stream to reproduce the symlinks
-            symlink = {
-              :name => name,
-              :mode => 0644,
-              :typeflag => '2',
-              :size => 0,
-              :linkname => File.readlink(p),
-              :prefix => '',
-            }
-            stream.write(Minitar::PosixHeader.new(symlink))
-          else
-            File.open(p, 'rb') do |r|
-              writer.add_file_simple(
-                name, :mode => 0644, :size => File.size(r)
-              ) do |d, _opts|
-                IO.copy_stream(r, d)
+        look_at = ['']
+        while (prefix = look_at.pop)
+          Dir.glob("#{prefix}**/*", File::FNM_DOTMATCH) do |p|
```

Comment:
1. Why do you want to match hidden files? Cookbooks can't be named with a leading `.`
2. You're dutifully using `File::SEPARATOR` everywhere... except you have a hard-coded slash here. Wat

Reply:
`Find.find` matches hidden files, so this is keeping parity. If something shouldn't be uploaded, it should be in chefignore. I think we make use of this e.g. `fb_skel/files/default/skel-e/.bashrc`.

I'll fix the `File::SEPARATOR`.